src/server/frontend_wayland/xdg_activation_v1.cpp

```diff
@@ -329,13 +260,18 @@ void mf::XdgActivationV1::Instance::activate(std::string const& token, struct wl
     // 2. The surface failed to use the token in the alotted period of time
     // 3. A key was pressed down between the issuing of the token and the activation
     //    of the surface with that token
+    //
     auto xdg_token = xdg_activation_v1->try_consume_token(token);
     if (!xdg_token)
     {
         mir::log_error("XdgActivationV1::activate invalid token: %s", token.c_str());
         return;
     }
 
+    // try_consume_token removes the token from our list of tokens, need to
+    // inform the token authority as well.
+    xdg_activation_v1->token_authority->revoke_token(token);
```

Comment:
Maybe a better approach would be to consume the token on the token authority side and  return it, which would call the "revocation callback" and remove it from xdg activation as well

Reply:
Can't work because we need the xdg-activation specific data. So we need to first get that, then request the token authority to remove the token (which removes the extra xdg-activation specific data as well)